core/src/main/java/io/grpc/CallOptions.java

```diff
@@ -370,7 +371,7 @@ public String toString() {
     toStringHelper.add("affinity", affinity);
     toStringHelper.add("executor", executor != null ? executor.getClass() : null);
     toStringHelper.add("compressorName", compressorName);
-    toStringHelper.add("customOptions", Arrays.toString(customOptions));
+    toStringHelper.add("customOptions", Arrays.deepToString(customOptions));
```

Comment:
@dapengzhang0 Hmm. I assume it was implemented so that `withOption` doesn't have to always iterate through the whole array, and that it's probably unlikely that the same Key gets overwritten several times. Not sure whether this is a good tradeoff or premature optimization. I would assume that it's very fast to search through the array, as it's likely very small and search would just have to do `==` comparison. Maybe @elandau can comment on the implementation. I can't find any dicussion about it on the PR.